src/api/app/services/scm_status_reporter.rb

```diff
@@ -0,0 +1,49 @@
+class SCMStatusReporter
+  attr_accessor :payload, :scm_token, :state
+
+  def initialize(payload, scm_token, event = nil)
+    @payload = payload
+    @scm_token = scm_token
+
+    @state = event.nil? ? 'pending' : scm_final_state(event)
+  end
+
+  def call
```

Comment:
I still don't understand where you filter for the project/package. You only want to report for  a *specific* package in a *specific* project right?

Reply:
This is handled in the job.